File: 27. 移除元素/main.go
package main

import "fmt"

func main() {
	fmt.Println(removeElement([]int{4,2,0,2,2,1,4,4,1,4,3,2}, 4))
	// fmt.Println(removeElement([]int{3,3}, 3))
}

func removeElement(nums []int, val int) int {
	var ans = len(nums)
	for i := 0; i < ans; {
		if nums[i] == val {
			nums[i] = nums[ans - 1]
			ans--
		} else {
			i++
		}
	}
	return ans
}

File: 27. 移除元素/swift/main/main/main.swift
//
//  main.swift
//  main
//
//  Created by ByteDance on 4/24/25.
//

import Foundation

class Solution {
    func removeElement(_ nums: inout [Int], _ val: Int) -> Int {
        var left = 0
        var right = nums.count - 1

        while left <= right {
            if nums[left] == val {
                let tmp = nums[right]
                nums[right] = nums[left]
                nums[left] = tmp
                right -= 1
            } else {
                left += 1
            }
        }

        return right + 1
    }
}

